### PR TITLE
Strong name all projects in the repository

### LINKED ISSUE #738: Strong name the releases please
I see #308 was closed with a suggestion to use StrongNamer. Which is fine, but...

Microsoft's new library guidance (out yesterday!) specifically recommends you strong-name sign your open-source libraries, and just provide the private key in the repo: https://docs.microsoft.com/en-ca/dotnet/standard/library-guidance/strong-naming.

Its worth mentioning that of the dozen-odd nuget packages I'm referencing, this is the only one that isn't signed.

### PR DESCRIPTION
**Pull request by antonfirsov**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

# Let's get over it!
Fixes #738.

There are unsigned external packages in ImageSharp.Benchmarks.csproj (Colorful and Pfim to benchmark colorspaces and TGA against other implementations). I tried to hack them with [StrongNamer](https://github.com/dsplaisted/strongnamer) but the problem is that the BDN doesn't seem to apply the StrongNamer MSBuild targets in it's builds, so colorspace & Tga benchmarks will fail at runtime.

I ended up with a workaround that does not use StrongNamer. It's documented in `ImageSharp.Benchmarks.csproj` comments.

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -28,10 +28,9 @@
 
   <ItemGroup>
     <InternalsVisibleTo Include="DynamicProxyGenAssembly2" PublicKey="0024000004800000940000000602000000240000525341310004000001000100c547cac37abd99c8db225ef2f6c8a3602f3b3606cc9891605d02baa56104f4cfc0734aa39b93bf7852f7d9266654753cc297e7d2edfe0bac1cdcf9f717241550e0a7b191195b7667bb4f64bcb8e2121380fd1d9d46ad2d92d2d15605093924cceaf74c4861eff62abf69b9291ed0a340e113be11e6a7d3113e92484cf7045cc7" />
```
Comment (JimBobSquarePants) [inline src/Directory.Build.props]: `PublicKey="$(SixLaborsPublicKey)"`

File: Directory.Build.props [hunk 2]
```diff
@@ -95,8 +95,9 @@
       <!-- Contains RemoteExecutor. Taken from: https://github.com/dotnet/runtime/blob/master/NuGet.config -->
       https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json;
     </RestoreSources>
-    <SixLaborsPublicKey>002400000c8000009400000006020000002400005253413100040000010001000147e6fe6766715eec6cfed61f1e7dcdbf69748a3e355c67e9d8dfd953acab1d5e012ba34b23308166fdc61ee1d0390d5f36d814a6091dd4b5ed9eda5a26afced924c683b4bfb4b3d64b0586a57eff9f02b1f84e3cb0ddd518bd1697f2c84dcbb97eb8bb5c7801be12112ed0ec86db934b0e9a5171e6bb1384b6d2f7d54dfa97</SixLaborsPublicKey>
+    <SixLaborsPublicKey>00240000048000009400000006020000002400005253413100040000010001000147e6fe6766715eec6cfed61f1e7dcdbf69748a3e355c67e9d8dfd953acab1d5e012ba34b23308166fdc61ee1d0390d5f36d814a6091dd4b5ed9eda5a26afced924c683b4bfb4b3d64b0586a57eff9f02b1f84e3cb0ddd518bd1697f2c84dcbb97eb8bb5c7801be12112ed0ec86db934b0e9a5171e6bb1384b6d2f7d54dfa97</SixLaborsPublicKey>
```
Comment (JimBobSquarePants) [inline Directory.Build.props]: How did you know which one was correct?

File: src/Directory.Build.props (hunk 1 above)
Comment (antonfirsov) [reply src/Directory.Build.props]: This one is a different. It's for dynamically emited code by Moq. Worth a comment.

File: Directory.Build.props (hunk 2 above)
Comment (antonfirsov) [reply Directory.Build.props]: Fed the snk to this script:
https://stackoverflow.com/a/43897217

Comment (JimBobSquarePants) [reply Directory.Build.props]: Ah cool!

**Comment by antonfirsov** [comment]
The tests have been passed several times already. Merging this with admin superpower, since it's blocking #1134.

**Pull request merged**